src/index.ts

```diff
@@ -17,18 +19,22 @@ export class AgnosticCMSHarmonizerClient {
     this.clientParams = clientParams;
   }
 
-  protected async agnosticCmsInitialize(handler: Function): Promise<any> {
+  protected async agnosticCmsInitialize<T = unknown>(
+    handler: () => T,
+  ): Promise<T> {
     try {
-      return await handler();
+      return handler();
```

Comment:
If the `Promise` doesn't become resolved at this point the error handling is not going to work.
Maybe we should add a `should fail` test suite as you proposed previously.

Reply:
Thanks for flagging! Corrected in [751613b](https://github.com/onebeyond/agnostic-cms-harmonizer/pull/13/commits/751613bb32d799e883c840dbfdb328fab912ad2f)